packages/plugin-http-errors/http-errors.js

```diff
@@ -0,0 +1,194 @@
+ * @param {Array|Object|number} config.httpErrorCodes - Error codes to capture
+ * @param {number} config.maxResponseSize - Maximum response body size to capture
+ * @param {number} config.maxRequestSize - Maximum request body size to capture
+ * @param {Function} config.onHttpError - Callback for intercepting HTTP errors
+ * @returns {Object} Bugsnag plugin
+ */
+module.exports = (config = {}, global = window) => {
+  const {
+    httpErrorCodes = DEFAULT_HTTP_ERROR_CODES,
+    maxRequestSize = DEFAULT_MAX_REQUEST_SIZE,
+    onHttpError
+  } = config
+
+  // Normalize httpErrorCodes to an array
+  const normalizedStatusCodes = Array.isArray(httpErrorCodes) ? httpErrorCodes : [httpErrorCodes]
+
+  let restoreFunctions = []
+  const plugin = {
+    name: 'httpErrors',
+    load: (client) => {
+      // Try to get existing request tracker
+      let requestTrackerPlugin = client.getPlugin('requestTracker')
+      const redactedKeys = client._config.redactedKeys || []
+
+      // Auto-load request tracker if not present
+      if (!requestTrackerPlugin) {
+        try {
+          const { createRequestTrackerPlugin } = require('@bugsnag/request-tracker')
+          const trackerPlugin = createRequestTrackerPlugin([], global)
+          client._loadPlugin(trackerPlugin)
+          requestTrackerPlugin = client.getPlugin('requestTracker')
+        } catch (error) {
+          client._logger.warn('Failed to auto-load request tracker, using direct fetch patching:', error.message)
+        }
+      }
+
+      // Use shared request tracker if available
+      if (requestTrackerPlugin) {
+        const { fetchTracker, xhrTracker } = requestTrackerPlugin
+
+        if (fetchTracker) {
+          restoreFunctions.push(fetchTracker._restore)
+          fetchTracker.onStart((startContext) => {
+            return {
+              onRequestEnd: (endContext) => {
+                handleHttpError(startContext, endContext)
+              }
+            }
+          })
+        }
+        if (xhrTracker) {
+          restoreFunctions.push(xhrTracker._restore)
+          xhrTracker.onStart((startContext) => {
+            return {
+              onRequestEnd: (endContext) => {
+                handleHttpError(startContext, endContext)
+              }
+            }
+          })
+        }
+      }
+
+      function handleHttpError (startContext, endContext) {
+        // Check if we should capture this status code
+        if (!shouldCaptureStatusCode(normalizedStatusCodes, endContext.status)) {
+          return
+        }
+
+        // console.log({ startContext, endContext })
+
+        try {
+          // Extract request information
+          const url = startContext.url
+          const method = startContext.method
+          const domain = extractDomain(url)
+
+          // Redact query parameters in URL
+          const redactedUrl = redactQueryParameters(url, redactedKeys)
+          const redactedQueryParams = parseQueryParams(redactedUrl)
```

Comment:
Should the redaction take place before or after the `onHttpError` callback?

Reply:
that's a great question, actually. @tomlongridge what do you think?